src/share/classes/com/sun/javatest/regtest/tool/Tool.java

```diff
@@ -1203,6 +1219,17 @@ public int run() throws BadArgs, Fault, Harness.Fault, InterruptedException {
             reportDirArg = Path.of("JTreport");
         }
 
+        if (customMainWrapper != null) {
+            CustomMainWrapper cmw = CustomMainWrapper.getInstance(customMainWrapper, customMainWrapperPathArg);
+            testVMOpts.add("-D" + MainWrapper.MAIN_WRAPPER + "=" + customMainWrapper);
+            testVMOpts.addAll(cmw.getAdditionalVMOpts());
```

Comment:
Ah, OK, there's the possibility that the instance of the wrapper class might want to specify some VM options. That's somewhat circular.  https://en.wikipedia.org/wiki/Ouroboros